src/api/axios.ts

```diff
@@ -0,0 +1,28 @@
+import axios, { InternalAxiosRequestConfig, AxiosResponse } from 'axios';
+import { onRequest, onRequestError } from './requestInterceptor';
+import { onResponse, onResponseError } from './responseInterceptor';
+
+const apiClient = axios.create({
+    baseURL: import.meta.env.VITE_API_ENDPOINT as string,
```

Comment:
Is there any reason for giving name for  API base URL as VITE_API_ENDPOINT ?

Reply:
I have done that to indicate it's a Vite specific environment variable.